src/UI/Dependency/Dependency.tsx

```diff
@@ -87,6 +94,7 @@ export const DependencyProvider: React.FC<{
       authHelper: authHelper || new DummyAuthHelper(),
       archiveHelper: archiveHelper || new DummyArchiveHelper(),
       authController: authController || new DummyAuthController(),
+      useAuth: useAuth || defaultAuthContext,
```

Comment:
defaultAuthContext not needed anymore if it's for free in the initialization in the `authContext.ts` file

Reply:
`dependencies: Partial<Dependencies>` indicates that any of dependencies properties can be undefined so it's required for the sake of the typescript(I think also tests would react differently, but that I didn't check, and I'm not 100% sure)

Reply:
If you update` authContext.ts` based on the example I gave you, useAuth will never be undefined because it has a default initialization, which is technically the noAuth one.

Reply:
but undefined that is addressed here doesn't come from the `useAuth` itself but from the `Partial<Dependencies>` . `authContext` file got correct type